Mathlib/Algebra/Module/Defs.lean

```diff
@@ -326,18 +326,26 @@ section
 variable (R)
 
 /-- `nsmul` is equal to any other module structure via a cast. -/
-theorem nsmul_eq_smul_cast (n : ℕ) (b : M) : n • b = (n : R) • b := by
+lemma natCast_smul_eq_nsmul (n : ℕ) (b : M) : (n : R) • b = n • b := by
   induction' n with n ih
   · rw [Nat.cast_zero, zero_smul, zero_smul]
   · rw [Nat.cast_succ, add_smul, add_smul, one_smul, ih, one_smul]
 
+/-- `nsmul` is equal to any other module structure via a cast. -/
+lemma ofNat_smul_eq_nsmul (n : ℕ) [n.AtLeastTwo] (b : M) : (OfNat.ofNat n : R) • b = n • b :=
```

Comment:
This should:
* Have `no_index` around the `ofNat`, and the usual library note
* Have `ofNat` on the RHS as well.

Reply:
Yep, of course